sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/interface.scala

```diff
@@ -249,6 +251,14 @@ case class CatalogTable(
 
   import CatalogTable._
 
+  /** Record all timeline between the table and the metastore. */
+  private val _metastoreOpsPhaseSummaries: ArrayBuffer[PhaseSummary] = ArrayBuffer.empty
+
+  def metastoreOpsPhaseSummaries: Seq[PhaseSummary] = _metastoreOpsPhaseSummaries.toSeq
```

Comment:
using table metadata to carry this information is not a good idea. Can we track it at the caller side?

Reply:
```
using table metadata to carry this information is not a good idea.
```
Agree, as the catalog table case class used as the cache key, I also hit some problem before
f82b355 b908ecb.
```
Can we track it at the caller side?
```
How about tracking them by QueryPlanningTracker, and use the case class as the key of the phase map.

Reply:
the key is the phase name, isn't it?

Reply:
The key should be catalog table, it's the same problem with https://github.com/apache/spark/pull/23327#issuecomment-449959400. I'll address this after the conclusion made.